src/lib.rs

```diff
@@ -232,6 +232,15 @@ pub trait Outgoing {
     type Incoming;
 }
 
+/// Gives users the ability to define their own serializable/deserializable errors.
+pub trait EndpointError {
+    /// The specific Error type when deserialization fails.
+    type E;
+    /// 
+    fn try_from_error(self, response: http::Response<Vec<u8>>)
+        -> Result<Self::E, error::ResponseDeserializationEndpointError>;
```

Comment:
… and use `Self` here.

```suggestion
        -> Result<Self, error::ResponseDeserializationEndpointError>;
```